sim/battle.ts

```diff
@@ -2553,6 +2553,18 @@ export class Battle {
 				}
 			}
 			break;
+		case 'revivalblessing':
+			action.pokemon.side.pokemonLeft++;
+			if (action.target.position < action.pokemon.side.active.length) action.target.isActive = true;
```

Comment:
```suggestion
			if (action.target.position < action.pokemon.side.active.length) {
						action.target.isActive = true;
						this.queue.cancelMove(action.target);
			}
```
Probably good to fix this bug while we're at it.

Reply:
Actually found another bug with this, in that the Pokemon should actually be switching in rather than being set to active. I tested this with an intimidate pokemon, and reviving it while it was in active slot made intimidate reactivate.

Reply:
It also happens after the revive message is shown so the placement of this needs to be moved.

Reply:
Oh I know what you're talking about. The same thing happens with Revives in previous gens too (going back to at least gen 5 where it was possible to use one via Wonder Launcher). I will implement, thanks